db/db.go

```diff
@@ -0,0 +1,58 @@
+package db
+
+//db global storage in memory
+var db = make(map[int64]interface{})
+
+//id auto incremental value
+var index int64 = 0
+
+//Entity is an interface that allows to interact with the database
+type Entity struct{}
+
+//Create a new record in the database
+func (e Entity) Create(ent interface{}) interface{} {
```

Comment:
Although accepting an interface{} as parameter is a good approach, returning an interface{} is a bit more complicated.

The best way to understand why returning an interface is not the best move is to realize that, whoever uses this, won't know what is it that you are giving to them.